app/actions/ConfigActionCreators.js

```diff
@@ -0,0 +1,25 @@
+import constants from '../constants';
+import ConfigAPI from '../api/ConfigAPI';
+
+export default class ConfigActionCreators {
+  static getAppData() {
+    return (dispatch) => {
+      dispatch({ type: constants.FETCH_APP_DATA });
+    }
+  }
+  static fetchConfigData() {
```

Comment:
I would transform this one function into 4 functions, 3 that aren't exported and one (this one) that is:

```js
const fetchConfigData = () => ({
  type: constants.FETCH_APP_DATA
});

const receiveConfigDataSuccess = (config) => ({
  type: constants.RECEIVE_CONFIG_DATA_SUCCESS, payload: config
});

const receiveConfigDataError = (error) => ({
  type: constants.RECEIVE_CONFIG_DATA_ERROR, error
})

export const fetchConfigData = () => (
  (dispatch) => {
    fetchConfigData();

    ConfigAPI.fetchConfig()
      .then(receiveConfigDataSuccess)
      .catch(receiveConfigDataError);
  }
)
```

So 1) the actions are factored out into separate functions. I think this makes thing clearer.

Then 2) instead of overloading `RECEIVE_CONFIG_DATA` to take both success and error. Have two separate actions for success & error. Then the reducer will handle both actions appropriately.